lib/parallel_tests/cucumber/scenarios.rb

```diff
@@ -10,22 +11,27 @@ module Cucumber
     class Scenarios
       class << self
         def all(files, options={})
-          tags = []
-          tags.concat options[:ignore_tag_pattern].to_s.split(/\s*,\s*/).map {|tag| "~#{tag}" }
-          tags.concat options[:test_options].to_s.scan(/(?:-t|--tags) (~?@[\w,~@]+)/).flatten
+          # Combine and generate a tag expression for given test options and ignore tag pattern. Refer here to understand how new tag expression syntax works - https://github.com/cucumber/cucumber/tree/master/tag-expressions
+          tags = ''
+          tags << options[:test_options].to_s unless options[:test_options].nil?
+          tags << " and " if !options[:ignore_tag_pattern].nil? && !options[:test_options].nil? # only add an 'and' operator if both test options and ignore tag pattern is passed
+          tags << "not (#{options[:ignore_tag_pattern].to_s})" unless options[:ignore_tag_pattern].nil?
 
-          split_into_scenarios files, tags.uniq
+          split_into_scenarios files, tags
         end
 
         private
 
-        def split_into_scenarios(files, tags=[])
-
-          # Create the tag expression instance from gherkin, this is needed to know if the scenario matches with the tags invoked by the request
-          tag_expression = ::Cucumber::Core::Gherkin::TagExpression.new(tags)
+        def split_into_scenarios(files, tags='')
 
+          # Create the tag expression instance from cucumber tag expressions parser, this is needed to know if the scenario matches with the tags invoked by the request
           # Create the ScenarioLineLogger which will filter the scenario we want
-          scenario_line_logger = ParallelTests::Cucumber::Formatters::ScenarioLineLogger.new(tag_expression)
+          if tags.empty?
+            scenario_line_logger = ParallelTests::Cucumber::Formatters::ScenarioLineLogger.new
+          else
+            tag_expression = ::Cucumber::TagExpressions::Parser.new.parse(tags)
+            scenario_line_logger = ParallelTests::Cucumber::Formatters::ScenarioLineLogger.new(tag_expression)
+          end
```

Comment:
```Ruby
args = []
args << ::Cucumber::TagExpressions::Parser.new.parse(tags) if tags.any?
scenario_line_logger = ParallelTests::Cucumber::Formatters::ScenarioLineLogger.new(*args)
```

Reply:
`tags` is a `String` now so we cannot use `.any?`. Also, we need to parse the tag expression string if it is not empty and not when it is empty. I made adjustments for those two, I updated the code as you suggested like:

```
args = []
args << ::Cucumber::TagExpressions::Parser.new.parse(tags) unless tags.empty?
scenario_line_logger = ParallelTests::Cucumber::Formatters::ScenarioLineLogger.new(*args)
```